Cognite.Simulator.Utils/SimulationRunnerBase.cs

```diff
@@ -530,6 +536,17 @@ async void PublishSimulationRunStatus(string runStatus, CancellationToken token)
                 configObj,
                 samplingRange,
                 token).ConfigureAwait(false);
+
+            await EndSimulationRun(simEv, startTime, modelState, configState, configObj, metadata, token);
+        }
+
+        protected virtual async Task EndSimulationRun(SimulationRunEvent simEv,
+            DateTime startTime,
+            T modelState,
+            U configState,
+            V configObj,
+            Dictionary<string, string> metadata,
```

Comment:
When is the metadata added to the event?

Reply:
So, since this case of adding metadata to the event was only prosper specific, we override this method in SimConnect here: https://github.com/cognitedata/simconnect-dotnet/pull/148/files#diff-d213be91332c1d1077ea1a1182c3fe747ff008d03b2815b681f3716229a85519R49

Reply:
We can do it as a workaround, but needs to be documented here. Considering that this code is for a library, it is an anti-pattern to have unused parameters in non-abstract methods.
Also consider if adding meta-data is prosper specific, as it could be a generic mechanism to customize the output events for all connectors.

Reply:
Good feedback, Ive made the `EndSimulationRun` an abstract with only one parameter which is the simulation event, it will be called at the end of a simulation run and any connector that wants to update their event metadata or do anything else can override it